### PR TITLE
bpo-45873: Get rid of bootstrap_python

### PR DESCRIPTION
**Pull request by gvanrossum**
Instead we use `$(PYTHON_FOR_REGEN) .../deepfreeze.py` with the
frozen .h file as input, as we did for Windows in [bpo-45850](https://bugs.python.org/issue45850).

We also get rid of the code that generates the .h files
when make regen-frozen is run (i.e., .../make_frozen.py),
as well as the generated MANIFEST file
(the latter depended on those .h files existing).

https://bugs.python.org/issue45873

### CODE REVIEW

File: PCbuild/find_python.bat [hunk 1]
```diff
@@ -31,13 +31,13 @@
...
+@if exist "%_Py_EXTERNALS_DIR%\pythonx86\tools\python.exe" ("%_Py_EXTERNALS_DIR%\pythonx86\tools\python.exe" -Ec "import sys; assert sys.version_info[:2] >= (3, 8)" >nul 2>nul) && (set PYTHON="%_Py_EXTERNALS_DIR%\pythonx86\tools\python.exe") && (set _Py_Python_Source=found in externals directory) && goto :found || rmdir /Q /S "%_Py_EXTERNALS_DIR%\pythonx86"
 
 @rem If HOST_PYTHON is recent enough, use that
-@if NOT "%HOST_PYTHON%"=="" @%HOST_PYTHON% -Ec "import sys; assert sys.version_info[:2] >= (3, 10)" >nul 2>nul && (set PYTHON="%HOST_PYTHON%") && (set _Py_Python_Source=found as HOST_PYTHON) && goto :found
+@if NOT "%HOST_PYTHON%"=="" @%HOST_PYTHON% -Ec "import sys; assert sys.version_info[:2] >= (3, 8)" >nul 2>nul && (set PYTHON="%HOST_PYTHON%") && (set _Py_Python_Source=found as HOST_PYTHON) && goto :found
 
 @rem If py.exe finds a recent enough version, use that one
-@for %%p in (3.10) do @py -%%p -EV >nul 2>&1 && (set PYTHON=py -%%p) && (set _Py_Python_Source=found %%p with py.exe) && goto :found
+@for %%p in (3.10 3.9 3.8) do @py -%%p -EV >nul 2>&1 && (set PYTHON=py -%%p) && (set _Py_Python_Source=found %%p with py.exe) && goto :found
```
Comment (zware) [inline PCbuild/find_python.bat:40]: Considering how at least one (my :( ) buildbot happily used 3.1 for `py -3.10`, we might want to put 3.10 at the end of the list here.

**Comment by gvanrossum** [comment]
(Sorry for the confusing push history. I had my branches mixed up. Should be ready for review now.)

**Review by zware — APPROVED** [review]
One minor suggestion, otherwise LGTM

File: PCbuild/find_python.bat (hunk 1 above)
Comment (gvanrossum) [reply PCbuild/find_python.bat:40]: Hm... Before 1037ca5a8ea001bfa2a198e08655620234e9befd this said `(3.9 3.8)` so I assume the intention was to try the latest first. Maybe @zooba can help out? (Steve, why would an older `py -3.10` on Windows 8.1 find 3.1? (Yes, 3.1 was installed there; and 3.10 wasn't. But it feels like an argument parsing issue in launcher.c.)

**Comment by gvanrossum** [comment]
About the failing test. This is (again :-( ) about an out-of-tree build. Here's the thing. For each module foo.py that's to be frozen there's a rule that creates a frozen foo.h from it, and then a rule that creates a deep-frozen foo.c from that. So far so good. Here's the weirdness. The freezing rule that creates foo.h writes to `$(srcdir)`, e.g.:
```
Python/frozen_modules/io.h: Programs/_freeze_module Lib/io.py
	Programs/_freeze_module io $(srcdir)/Lib/io.py $(srcdir)/Python/frozen_modules/io.h
```
This seems inconsistent since `$(srcdir)` is not mentioned in the target (the first line), only in the command (the second line). And the deep-freezing rule that creates foo.c reads from `$(srcdir)` and writes to the build dir:
```
Python/deepfreeze/io.c: $(srcdir)/Python/frozen_modules/io.h $(DEEPFREEZE_DEPS)
	$(PYTHON_FOR_REGEN) $(srcdir)/Tools/scripts/deepfreeze.py $(srcdir)/Python/frozen_modules/io.h -m io -o Python/deepfreeze/io.c
```
I could go on forever randomly adding or removing `$(srcdir)` from one place or the other, but does anyone have any idea what this *should* do?

Note that there's one more mention of `foo.h`, namely in the definition of `FROZEN_FILES_OUT`. This does not use `$(srcdir)`.

**Comment by tiran** [comment]
The rule of thumb for for ``$(srcdir)`` is: any file in git gets a ``$(srcdir)`` prefix and any generated file does not have a prefix. The file ``Python/frozen_modules/io.h`` is not committed to git and generated by a different rule. Therefore it does not get a ``$(srcdir)`` prefix.

The idea behind out-of-tree (OOT) builds is a separation of source files and target files. Source files are any files we ship to users, either in a git checkout or in a source tarball. OOT building treat the source directory is an immutable, read-only source and places any generated file in a separate directory structure. The feature is also known as *out of source build* or *autoconf with VPATH*.

## Example for separate debug and release builds

```
cd cpthon
make clean
mkdir -p builddir/release
mkdir -p builddir/debug
```

```
cd builddir/release
../../configure -C
make
```

```
cd ../debug
../../configure -C --with-pydebug
make
```

File: Tools/scripts/deepfreeze.py [hunk 2]
```diff
@@ -86,12 +87,17 @@ def analyze_character_width(s: str) -> tuple[int, bool]:
     return kind, ascii
 
 
+def removesuffix(base, suffix):
```
Comment (kumaraditya303) [inline Tools/scripts/deepfreeze.py]:
```suggestion
def removesuffix(base: str, suffix: str) -> str:
```

**Pull request merged**